### PR TITLE
Ref: make hints optional

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring

## :scroll: Description
Ref: make hints optional

## :bulb: Motivation and Context
We should use the power of the language instead of forcing users to pass null

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: flutter/lib/src/default_integrations.dart [hunk 1]
```diff
@@ -47,7 +47,7 @@ Integration loadContextsIntegration(
 ) {
   Future<void> integration(Hub hub, SentryOptions options) async {
     options.addEventProcessor(
-      (event, dynamic hint) async {
+      (event, {hint}) async {
```
Comment (rxlabz) [inline flutter/lib/src/default_integrations.dart:50]:
```suggestion
      (event, {dynamic hint}) async {
```
we should avoid implicit dynamic

File: flutter/lib/src/default_integrations.dart [hunk 2]
```diff
@@ -147,7 +147,7 @@ Integration loadAndroidImageListIntegration(
...
```
Comment (rxlabz) [inline flutter/lib/src/default_integrations.dart:150]: https://github.com/getsentry/sentry-dart/pull/210/files#r533360724

File: flutter/lib/src/default_integrations.dart (hunk 1 above)
Comment (marandaneto) [reply flutter/lib/src/default_integrations.dart:50]: wondering if the `event` should be typed too then?

**Review by rxlabz — APPROVED** [review]
LGTM

**Pull request merged**